source/common/config/delta_subscription_state.cc

```diff
@@ -157,6 +162,28 @@ void DeltaSubscriptionState::handleEstablishmentFailure() {
 
 envoy::api::v2::DeltaDiscoveryRequest DeltaSubscriptionState::getNextRequest() {
   envoy::api::v2::DeltaDiscoveryRequest request;
+  if (!any_request_sent_yet_in_current_stream_) {
+    populateDiscoveryRequest(request);
+  } else if (!aliases_added_.empty()) {
+    populateDiscoveryRequestWithAliases(request);
+  } else {
+    populateDiscoveryRequest(request);
+  }
+
+  request.set_type_url(type_url_);
+  request.mutable_node()->MergeFrom(local_info_.node());
+  return request;
+}
+
+void DeltaSubscriptionState::populateDiscoveryRequestWithAliases(
+    envoy::api::v2::DeltaDiscoveryRequest& request) {
+  std::copy(aliases_added_.begin(), aliases_added_.end(),
+            Protobuf::RepeatedFieldBackInserter(request.mutable_resource_names_subscribe()));
+  aliases_added_.clear();
+}
+
+void DeltaSubscriptionState::populateDiscoveryRequest(
+    envoy::api::v2::DeltaDiscoveryRequest& request) {
```

Comment:
You may be able to factor out the `if (!any_request_sent_yet_in_current_stream_)` block from populateDiscoveryRequest into its own function. Then, rather than having ifs+elses in getNextRequest(), you could just do
```
if (!any_request_sent_yet_in_current_stream_)
  prepareFirstRequest(request);
if (!aliases_added_.empty())
  populateWithAliases(request);
populateDiscoveryRequest(request);
```